src/symbolic/scoped_symbol.ml

```diff
@@ -0,0 +1,107 @@
+type scope_token =
+  | Open_scope of string
+  | Close_scope
+  | Symbol of Smtml.Symbol.t
+
+let pp fmt token =
+  match token with
+  | Open_scope name -> Fmt.pf fmt "open %s" name
+  | Close_scope -> Fmt.pf fmt "close"
+  | Symbol _ -> Fmt.pf fmt "symbol"
+
+let pp_list fmt tokens =
+  Fmt.list ~sep:(fun fmt () -> Fmt.pf fmt " ") pp fmt tokens
+
+let symbol s = Symbol s
+
+let only_symbols tokens =
+  List.filter_map (function Symbol s -> Some s | _ -> None) tokens
```

Comment:
`| _ -> None` should be replaced by `| Open_scope _ | Close_scope -> None` so that we don't get a bug if we ever add a new case here later (sorry, I did not think about it when we wrote that together)